### DISCUSSION TITLE
Copy all Infolists TextEntry's to clipboard at once

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#14120] [Help] [answered]
### Package

Infolist builder

### Package Version

v3.2.110

### How can we help you?

Expected Result:

halloween, halloween decor, halloween shirt

my current `->action` in `copy_all_keywords` it give me only `, , ,` without the copied states

**My Current Code:**

```
public function infolist(Infolist $infolist): Infolist
{
    return $infolist
        ->state([
            'keyword_1' => 'halloween',
            'keyword_2' => 'halloween decor',
            'keyword_3' => 'halloween shirt',
        ])
        ->schema([
            Section::make(__('Keywords Results'))
                ->headerActions([
                    Action::make('copy_all_keywords')
                        ->action(function (Livewire $livewire, Section $component): void {
                            $keywords = collect([
                                $component->getState('keyword_1'),
                                $component->getState('keyword_2'),
                                $component->getState('keyword_3'),
                            ])->implode(', ');

                            dd($keywords);

                            $livewire->js('window.navigator.clipboard.writeText("' . $keywords . '");');
                        }),
                ])
                ->schema([
                    TextEntry::make('keyword_1'),

                    TextEntry::make('keyword_2'),

                    TextEntry::make('keyword_3'),
                ]),
        ]);
}
```

**Comment by leandrocfe** [comment, marked as answer]
```php
$state = [
    'keyword_1' => 'halloween',
    'keyword_2' => 'halloween decor',
    'keyword_3' => 'halloween shirt',
];

return $infolist
    ->state($state)
    ->schema([
        Section::make(__('Keywords Results'))
            ->headerActions([
                Action::make('copy_all_keywords')
                    ->alpineClickHandler(fn () => '
                        window.navigator.clipboard.writeText("'.implode(', ', $state).'");
                        $tooltip(\'Copied\', {
                            theme: $store.theme,
                            timeout: 2000,
                        })
                    '),
            ])
            ->schema([
                TextEntry::make('keyword_1'),
                TextEntry::make('keyword_2'),
                TextEntry::make('keyword_3'),
            ]),
    ]);
```

**Reply by devhoussam1998** [reply]
How about this?

```
public function table(Table $table): Table
    {
        return $table
            ->query(User::query())
            ->columns([])
            ->filters([])
            ->actions([
                ViewAction::make()
                    ->infolist([
                        InfolistSection::make(__('User Details'))
                            ->headerActions([
                                InfolistAction::make('copy')
                                    ->action(function (): void {
                                        Notification::make()
                                            ->success()
                                            ->title(__('Copied Successfully'))
                                            ->send();
                                    }),
                            ])
                            ->schema([
                                TextEntry::make('name'),
                            ]),
            ]);
    }
```

**Reply by leandrocfe** [reply]
what is the issue?

**Reply by devhoussam1998** [reply]
> what is the issue?

`Undefined variable $state`

```
public function table(Table $table): Table
    {
        return $table
            ->query(User::query())
            ->columns([])
            ->filters([])
            ->actions([
                ViewAction::make()
                    ->infolist([
                        InfolistSection::make(__('User Details'))
                            ->headerActions([
                                InfolistAction::make('copy')
                                    ->action(function (): void {
                                        Notification::make()
                                            ->success()
                                            ->title(__('Copied Successfully'))
                                            ->send();
                                    }),
                            ])
                            ->schema([
                                TextEntry::make('name'),
                            ]),
            ]);
    }
```

**Reply by leandrocfe** [reply]
> > what is the issue?
> 
> `Undefined variable $state`
> 
> ```
> public function table(Table $table): Table
>     {
>         return $table
>             ->query(User::query())
>             ->columns([])
>             ->filters([])
>             ->actions([
>                 ViewAction::make()
>                     ->infolist([
>                         InfolistSection::make(__('User Details'))
>                             ->headerActions([
>                                 InfolistAction::make('copy')
>                                     ->action(function (): void {
>                                         Notification::make()
>                                             ->success()
>                                             ->title(__('Copied Successfully'))
>                                             ->send();
>                                     }),
>                             ])
>                             ->schema([
>                                 TextEntry::make('name'),
>                             ]),
>             ]);
>     }
> ```
> 

Where is $state in this code?

**Reply by devhoussam1998** [reply]
> > > what is the issue?
> > 
> > 
> > `Undefined variable $state`
> > ```
> > public function table(Table $table): Table
> >     {
> >         return $table
> >             ->query(User::query())
> >             ->columns([])
> >             ->filters([])
> >             ->actions([
> >                 ViewAction::make()
> >                     ->infolist([
> >                         InfolistSection::make(__('User Details'))
> >                             ->headerActions([
> >                                 InfolistAction::make('copy')
> >                                     ->action(function (): void {
> >                                         Notification::make()
> >                                             ->success()
> >                                             ->title(__('Copied Successfully'))
> >                                             ->send();
> >                                     }),
> >                             ])
> >                             ->schema([
> >                                 TextEntry::make('name'),
> >                             ]),
> >             ]);
> >     }
> > ```
> 
> Where is $state in this code?

I dont understand?
